lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3768,7 +3762,14 @@ void ByteCodeGenerator::StartEmitFunction(ParseNode *pnodeFnc)
         else
         {
             Symbol *sym = funcInfo->root->sxFnc.GetFuncSymbol();
-            funcInfo->bodyScope->AddSymbol(sym);
+            if (funcInfo->paramScope->GetCanMergeWithBodyScope())
```

Comment:
The merge happens before this so I cannot add to param scope in non-split scope case.